### PR TITLE
Update mod to Minecraft 1.17

### PR DESCRIPTION
**Pull request by haykam821**
This pull request updates Traverse to Minecraft 1.17.

### CODE REVIEW

**Comment by NeusFear** [comment]
Looks good so far. What's the status?

**Comment by haykam821** [comment]
I think this pull request is ready for review. I believe the remaining issues are just boats and signs (caused by Terraform) and potential world generation differences compared to 1.16.

**Comment by NeusFear** [comment]
I'd like to see the Terraform issues fixed before approval/merge. I haven't found any issues with the code, just want to make sure everything is working.

**Comment by Prospector** [comment]
I've fixed a few world gen issues I found, have updated to the latest 1.16 changes, and also disabled the fir sign since it's broken. I think this is fine to release as a beta.

File: src/main/resources/data/traverse/recipes/fir_boat.json [hunk 1]
```diff
@@ -11,6 +11,6 @@
     }
   },
   "result": {
-    "item": "traverse:fir_boat"
+    "item": "minecraft:oak_boat"
```
Comment (haykam821) [inline src/main/resources/data/traverse/recipes/fir_boat.json:14]: This is incorrect.

Comment (Prospector) [reply src/main/resources/data/traverse/recipes/fir_boat.json:14]: whoops, thanks for noticing

**Pull request merged**